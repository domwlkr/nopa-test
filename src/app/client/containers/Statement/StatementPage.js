import React, {PropTypes} from 'react';
import { connect } from 'react-redux';
import { Layout, StatementContent } from '../../components';
import { getTransactions, getTransactionsSuccess, getTransactionsFail } from '../../redux/actions/transactionActions';
import { browserHistory } from 'react-router';

class StatementPage extends React.Component {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    // if (!this.props.loggedIn) browserHistory.push('/');

    this.props.getTransactions();
  }

  showMore() {

  }

  render() {
    return (
      <Layout title="Statement">
        <StatementContent onClickshowMore={this.showMore}
          bank={this.props.bank}
          accountDetails={this.props.accountDetails}
          transactions={this.props.transactions}
          loading={this.props.loading} />
      </Layout>
    );
  }
}

StatementPage.propTypes = {
  accountDetails: PropTypes.object,
  loading: PropTypes.bool,
  bank: PropTypes.object,
  showMore: PropTypes.func,
  transactions: PropTypes.object,
  getTransactions: PropTypes.func
};

const mapStateToProps = (state) => {
  return {
    transactions: state.transactionReducer.transactions,
    loading: state.transactionReducer.loading,
    accountDetails: state.formReducer.form.formData,
    loggedIn: state.formReducer.form.loggedIn,
    bank: state.bankReducer.bank
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getTransactions: () => {
      dispatch(getTransactions()).then((response) => {
        let data = response.payload.data ? response.payload.data.transactions : [{errors: {data : 'Network Error'}}];

        !response.error ? dispatch(getTransactionsSuccess(data)) : dispatch(getTransactionsFail(data));
      });
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(StatementPage);
