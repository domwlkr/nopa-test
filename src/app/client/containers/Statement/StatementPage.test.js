import React from 'react';
import StatementPage from './StatementPage';
import { Provider } from 'react-redux';
import configureStore from '../../redux/store/configureStore';
import renderer from 'react-test-renderer';

const store = configureStore();

test('Renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}><StatementPage /></Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
