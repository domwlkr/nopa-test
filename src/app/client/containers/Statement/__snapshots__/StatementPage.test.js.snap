// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<section
  className={undefined}
>
  <div
    className="header"
  >
    <img
      className="logo"
      src="test-file-stub"
    />
    <div
      className="login"
    >
      <a
        className="button"
        onClick={[Function]}
        style={Object {}}
      >
        Log In
      </a>
    </div>
  </div>
  <div
    className="layout"
  >
    <div
      className="main-content"
    >
      <div
        className="account-details"
      >
        <div
          className="account-details__left"
        >
          <span />
          <span />
        </div>
        <div
          className="account-details__right"
        >
          <span>
            Current account
          </span>
          <span />
          <span />
        </div>
      </div>
      <div
        className="table-wrap"
      >
        <p>
          Your transactions for the last 30 days
        </p>
        <div
          className="transactions-table"
        />
      </div>
      <a
        className="button"
        onClick={[Function]}
        style={Object {}}
      >
        Show more
      </a>
    </div>
  </div>
  <section
    className="partners"
  >
    <div>
      <div>
        <span>
          Our partners:
        </span>
      </div>
      <div>
        <div
          className="partner"
        >
          <img
            alt="Airbnb"
            src="test-file-stub"
          />
        </div>
        <div
          className="partner"
        >
          <img
            alt="Metro"
            src="test-file-stub"
          />
        </div>
        <div
          className="partner"
        >
          <img
            alt="Pariti"
            src="test-file-stub"
          />
        </div>
        <div
          className="partner"
        >
          <img
            alt="Unshackled"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </section>
  <footer
    className="footer"
  >
    <div>
      <p>
        © Nopa Limited 2017 All rights reserved. ‘Nopa’ and the Nopa logo are trade marks of Nopa Limited. Nopa is a member of CIFAS - the UK's leading anti-fraud association, and we are registered with the Office of the Information Commissioner (No. Z879078).
      </p>
      <p>
        Nopa Limited is incorporated in England & Wales (registration number 05197592), with its registered office at 1st Floor, Cottons Centre, Tooley Street, London, SE1 2QG. Nopa Limited is authorised and regulated by the Financial Conduct Authority, and entered on the Financial Services Register under firm registration number 563134.
      </p>
    </div>
  </footer>
</section>
`;
